workflow/scripts/savetoh5.jl

```diff
@@ -0,0 +1,176 @@
+├─ 🏷️ reference
+├─ 📂 floe_properties
+│  ├─ 🏷️ Description of labeled_image
+│  ├─ 🏷️ Description of properties
+│  ├─ 🔢 colunm_names
+│  ├─ 🔢 labeled_image
+│  └─ 🔢 properties
+└─ 📂 index
+   ├─ 🔢 latitude
+   ├─ 🔢 longitude
+   ├─ 🔢 time
+   ├─ 🔢 x
+   └─ 🔢 y
+```
+# The `floe_properties` and `index` group
+
+The `floe_properties` group contains a floe properties matrix `properties` for `labeled_image` and assciated `colunm_names`.
+
+The `index` group contains the latitude and longitude coordinates (see the description of properties within the file for an account of the units of each property), and satellite pass time `time` in Unix time that captured the source image.
+
+"""
+function makeh5file(pathtosampleimg, resdir)
+
+    latlondata = getlatlon(pathtosampleimg)
+
+    iftversion = getiftversion()
+
+    ptpath = joinpath(resdir, "passtimes.jls")
+    passtimes = deserialize(ptpath)
+    ptsunix = Int64.(Dates.datetime2unix.(passtimes))
+
+    fnpath = joinpath(resdir, "filenames.jls")
+    truecolor_refs, reflectance_refs = deserialize(fnpath)
+
+    floespath = joinpath(resdir, "segmented_floes.jls")
+    floes = deserialize(floespath)
+
+    colstodrop = [:row_centroid, :col_centroid, :min_row, :min_col, :max_row, :max_col]
+    propspath = joinpath(resdir, "floe_props.jls")
+    props = deserialize(propspath)
+    for df in props
+        converttounits!(df, latlondata, colstodrop)
+    end
+
+    h5dir = joinpath(resdir, "hdf5-files")
+    mkpath(h5dir)
+    for (i, fname) in enumerate(truecolor_refs)
+        fname = makeh5filename(fname)
+        fnamepath = joinpath(h5dir, fname)
+        h5open(fnamepath, "w") do file
+            # Add top-level attributes
+            attrs(file)["fname_reflectance"] = reflectance_refs[i]
+            attrs(file)["fname_truecolor"] = truecolor_refs[i]
+            attrs(file)["iftversion"] = iftversion
+            attrs(file)["crs"] = latlondata["crs"]
+            attrs(file)["reference"] = "https://doi.org/10.1016/j.rse.2019.111406"
+            attrs(file)["contact"] = "mmwilhelmus@brown.edu"
+
+            g = create_group(file, "index")
+            g["time"] = Int64.(Dates.datetime2unix(passtimes[i]))
+            g["x"] = latlondata["X"]
+            g["y"] = latlondata["Y"]
+            g["latitude"] = latlondata["latitude"]
+            g["longitude"] = latlondata["longitude"]
+
+            g = create_group(file, "floe_properties")
+            g["properties"] = Matrix(props[i])
+            attrs(g)["Description of properties"] = """Generated using the `regionprops` function from the `skimage` package. See https://scikit-image.org/docs/0.20.x/api/skimage.measure.html#regionprops
+
+            Area units (`area`, `convex_area`) are in sq. kilometers, and length units (`minor_axis_length`, `major_axis_length`, and `perimeter`) are in kilometers, and `orientation` in radians (see the description of properties attribute.). Latitude and longitude coordinates are in degrees.
```

Comment:
Need columns for "x" and "y" in stereographic coordinates in the property matrix also. The units would be m relative to the NSIDC north polar stereographic projection.